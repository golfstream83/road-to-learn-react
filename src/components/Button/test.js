import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button } from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
    const props = {
        onClick: () => '',
        className: 'button',
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Button { ...props }>Give Me More</Button>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Button { ...props }>Give Me More</Button>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('shows text of the button', () => {
        const element = shallow(
            <Button { ...props } >Button</Button>
        );
        expect(element.text()).toEqual('Button');
    });
});