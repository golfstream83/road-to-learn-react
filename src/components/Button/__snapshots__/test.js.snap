// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  Give Me More
</button>
`;
